src/kirby/components/icon/icon.component.ts

```diff
@@ -1,16 +1,66 @@
-import { Component, OnInit, Input, Output } from '@angular/core';
+import { Component, Input } from '@angular/core';
 
 @Component({
   selector: 'kirby-icon',
   templateUrl: './icon.component.html',
   styleUrls: ['./icon.component.scss'],
 })
-export class IconComponent implements OnInit {
-  @Input() name = '';
+export class IconComponent {
+  static DEFAULT_ICON_CODE = 0xf2cf;
+  static DEFAULT_ICON_NAME = 'cog';
 
-  spriteFile = '../../assets/icons/icon-sprite.svg';
+  @Input() name:
+    | string
+    | 'add'
+    | 'close'
+    | 'cog'
+    | 'swap'
+    | 'move'
+    | 'log-out'
+    | 'more'
+    | 'arrow-back'
+    | 'help'
+    | 'attach'
+    | 'search'
+    | 'checkbox-outline'
+    | 'checkbox'
+    | 'menu'
+    | 'person' = IconComponent.DEFAULT_ICON_NAME;
 
-  constructor() {}
+  @Input() size: string | 'small' | 'large' = 'small';
+  @Input() colorType: string | 'primary' | 'secondary' | 'tertiary' = 'primary';
 
-  ngOnInit() {}
+  getIonicIconName(name: string): string {
+    const icon = this.searchIcons(name);
+    return icon !== undefined ? icon.name : IconComponent.DEFAULT_ICON_NAME;
+  }
+
+  getCharCode(name: string): string {
+    const icon = this.searchIcons(name);
+    return icon !== undefined
+      ? String.fromCharCode(icon.code)
+      : String.fromCharCode(IconComponent.DEFAULT_ICON_CODE);
+  }
+
+  private searchIcons(name: string): any {
```

Comment:
Consider removing return type or make it explicit:
```suggestion
  private searchIcons(name: string): { code: number; name: string } {
```